sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+        for each feature subset size selected by the
+        sequential feature selection algorithm.
+        Here the dictionary keys are the lengths k of these feature subsets.
+        The dictionary values are dictionaries themselves with the following
+        keys: 'feature_subset_idx' (tuple of indices of the feature subset)
+              'cv_scores' (list individual cross-validation scores)
+              'avg_score' (average cross-validation score)
+
+    Examples
+    --------
+    The following example shows how to use the sequential feature selector
+    with default settings to select a feature subset, consisting of
+    1 to 3 features, from iris. The selection criteria for this
+    feature subset is the average cross-validation performance
+    (cv=5 by default) of the `estimator` (here: KNN)
+    during the greedy forward selection search.
+
+        >>> from sklearn.feature_selection import SequentialFeatureSelector
+        >>> from sklearn.neighbors import KNeighborsClassifier
+        >>> from sklearn.datasets import load_iris
+        >>> iris = load_iris()
+        >>> X, y = iris.data, iris.target
+        >>> knn = KNeighborsClassifier(n_neighbors=3)
+        >>> sfs = SequentialFeatureSelector(knn, n_features_to_select=(1, 3))
+        >>> sfs = sfs.fit(X, y)
+        >>> sfs.score_  # doctest: +ELLIPSIS
+        0.9733...
+        >>> sfs.feature_subset_idx_
+        (0, 2, 3)
+        >>> sfs.transform(X).shape
+        (150, 3)
+
+    See also
+    --------
+    RFE : Recursive feature elimination selection of the best
+    number of features
+
+    References
+    ----------
+    .. [1] Ferri, F. J., Pudil P., Hatef, M., Kittler, J., "Comparative
+           study of techniques for large-scale feature selection,"
+           Pattern Recognition in Practice IV : 403-413, 1994
+
+    """
+    def __init__(self, estimator, n_features_to_select=1,
+                 forward=True, scoring=None,
+                 cv=5, n_jobs=1,
+                 pre_dispatch='2*n_jobs'):
+
+        self.estimator = estimator
+        self.n_features_to_select = n_features_to_select
+        self.forward = forward
+        self.pre_dispatch = pre_dispatch
+        self.scoring = scoring
+        self.cv = cv
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """Perform feature selection and learn model from training data.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+
+        """
+
+        if not (isinstance(self.n_features_to_select, int) or
+                isinstance(self.n_features_to_select, tuple)):
+            raise ValueError('n_features_to_select must be a positive integer'
```

Comment:
Perhaps we should just make it easy and allow the user to specify a proportion of features

Reply:
Hm, I actually like the integers because they allow for being more explicit and less ambiguity (e.g., 50% of 15 features, people would have to read/check whether it would select 7 or 8 features)